src/common/open-api/ConductorClient.ts

```diff
@@ -0,0 +1,76 @@
+/* istanbul ignore file */
+/* tslint:disable */
+/* eslint-disable */
+import type { BaseHttpRequest } from './core/BaseHttpRequest';
+import type { OpenAPIConfig } from './core/OpenAPI';
+import { NodeHttpRequest } from './core/NodeHttpRequest';
+
+import { AdminResourceService } from './services/AdminResourceService';
+import { ApplicationResourceService } from './services/ApplicationResourceService';
+import { AuthorizationResourceService } from './services/AuthorizationResourceService';
+import { EventResourceService } from './services/EventResourceService';
+import { GroupResourceService } from './services/GroupResourceService';
+import { HealthCheckResourceService } from './services/HealthCheckResourceService';
+import { MetadataResourceService } from './services/MetadataResourceService';
+import { MigrationResourceService } from './services/MigrationResourceService';
+import { QueueAdminResourceService } from './services/QueueAdminResourceService';
+import { SchedulerResourceService } from './services/SchedulerResourceService';
+import { TaskResourceService } from './services/TaskResourceService';
+import { TokenResourceService } from './services/TokenResourceService';
+import { UserResourceService } from './services/UserResourceService';
+import { WorkflowBulkResourceService } from './services/WorkflowBulkResourceService';
+import { WorkflowResourceService } from './services/WorkflowResourceService';
+
+type HttpRequestConstructor = new (config: OpenAPIConfig) => BaseHttpRequest;
+
+export class ConductorClient {
+
+  public readonly adminResource: AdminResourceService;
+  public readonly applicationResource: ApplicationResourceService;
+  public readonly authorizationResource: AuthorizationResourceService;
+  public readonly eventResource: EventResourceService;
+  public readonly groupResource: GroupResourceService;
+  public readonly healthCheckResource: HealthCheckResourceService;
+  public readonly metadataResource: MetadataResourceService;
+  public readonly migrationResource: MigrationResourceService;
+  public readonly queueAdminResource: QueueAdminResourceService;
+  public readonly schedulerResource: SchedulerResourceService;
+  public readonly taskResource: TaskResourceService;
+  public readonly tokenResource: TokenResourceService;
+  public readonly userResource: UserResourceService;
+  public readonly workflowBulkResource: WorkflowBulkResourceService;
+  public readonly workflowResource: WorkflowResourceService;
+
+  public readonly request: BaseHttpRequest;
+
+  constructor(config?: Partial<OpenAPIConfig>, HttpRequest: HttpRequestConstructor = NodeHttpRequest) {
```

Comment:
I wonder if we could use this second param to do customizations for `tls` and/or auth.